apis/supervisor/idp/v1alpha1/types_githubidentityprovider.go.tmpl

```diff
@@ -0,0 +1,218 @@
+
+	// TLS configuration for GitHub Enterprise Server.
+	// +optional
+	TLS *TLSSpec `json:"tls,omitempty"`
+}
+
+// GitHubUsernameClaimSpec allows the user to specify which attribute(s) from GitHub to use for the downstream username.
+// See the response for https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user.
+type GitHubUsernameClaimSpec string
+
+const (
+	// GitHubUsernameId specifies using the `id` attribute from GitHub as the downstream username.
+	GitHubUsernameId GitHubUsernameClaimSpec = "id"
+
+	// GitHubUsernameLogin specifies using the `login` attribute from GitHub as the downstream username.
+	GitHubUsernameLogin GitHubUsernameClaimSpec = "login"
+
+	// GitHubUsernameLoginAndId specifies combining the `login` and `id` attributes from GitHub as the
+	// downstream username, separated by a colon. Example: "my-login:1234"
+	GitHubUsernameLoginAndId GitHubUsernameClaimSpec = "login:id"
+)
+
+// GitHubGroupNamesSpec allows the user to specify which attribute from GitHub to use for the downstream groups.
+// See the response for https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#list-teams-for-the-authenticated-user.
+type GitHubGroupNamesSpec string
+
+const (
+	// GitHubGroupsAsNames specifies using the `name` attribute from GitHub as the downstream group names
+	GitHubGroupsAsNames GitHubGroupNamesSpec = "name"
+
+	// GitHubGroupsAsSlugs specifies using the `slug` attribute from GitHub as the downstream group names
+	GitHubGroupsAsSlugs GitHubGroupNamesSpec = "slug"
+)
+
+// GitHubClaims allows customization of the username and groups claims
+type GitHubClaims struct {
+	// Username configures which property of the GitHub user record shall determine the username in Kubernetes.
+	//
+	// Can be either "id", "login", or "login:id". Defaults to "login:id".
+	//
+	// GitHub's user login attributes can only contain alphanumeric characters and non-repeating hyphens,
+	// and may not start or end with hyphens. GitHub users are allowed to change their login name,
+	// although it is inconvenient. If a GitHub user changed their login name from "foo" to "bar",
+	// then a second user might change their name from "baz" to "foo" in order to take the old
+	// username of the first user. For this reason, it is not as safe to make authorization decisions
+	// based only on the user's login attribute.
+	//
+	// If desired, an admin could configure identity transformation expressions on a FederationDomain to
+	// further customize how these usernames are presented to Kubernetes.
+	//
+	// Defaults to "login:id", which is the user login attribute, followed by a colon, followed by the unique and
+	// unchanging integer ID number attribute. This blends human-readable login names with the unchanging ID value
+	// from GitHub. Note that colons are not allowed in GitHub login attributes or ID numbers, so the colon in the
+	// "login:id" name will always be the login:id separator colon.
```

Comment:
so the colon in the "login:id" name will always be the colon as separator.

This might be more clear?

Reply:
done.